web/libs/editor/src/stores/AppStore.js

```diff
@@ -521,6 +528,19 @@ export default types
 
         if (res && res.then) res.then(resolve);
         else resolve(res);
+      }).then(() => {
+        self.setFlags({ isSubmittingDraft: false });
+      });
+    }
+
+    function waitForDraftSubmission() {
+      return new Promise((resolve) => {
```

Comment:
I think that it could be a bit more neat with promise in a model instead of flag, like
`self.draftInProgress = new Promise(...)` in `submitDraft()` above. And instead of `await self.waitForDraftSubmission()` we will just `await self.draftInProgress`.
But flag is really more consistent, so let's go with it!
Could you just please add a check for the flag at the beginning of `waitForDraftSubmission()` so we don't need to wait for 100ms if there is no draft in progress?